nobrainer/dataset.py

```diff
@@ -149,25 +151,25 @@ def get_dataset(
         dataset = dataset.map(lambda x, y: (normalizer(x), y))
 
     # Augment examples if requested.
-    if augment:
+    if augment is not None:
         if not scalar_label:
-            dataset = dataset.map(
-                lambda x, y: tf.cond(
-                    tf.random.uniform((1,)) > 0.5,
-                    true_fn=lambda: apply_random_transform(x, y),
-                    false_fn=lambda: (x, y),
-                ),
-                num_parallel_calls=num_parallel_calls,
-            )
+            for transform, kwargs in augment:
+                dataset = dataset.map(
+                    lambda x, y: tf.cond(
+                        tf.random.uniform((1,)) > 0.5,
+                        true_fn=lambda: (transform(x, **kwargs), y),
+                        false_fn=lambda: (x, y),
+                    ),
+                )
         else:
-            dataset = dataset.map(
-                lambda x, y: tf.cond(
-                    tf.random.uniform((1,)) > 0.5,
-                    true_fn=lambda: apply_random_transform_scalar_labels(x, y),
-                    false_fn=lambda: (x, y),
-                ),
-                num_parallel_calls=num_parallel_calls,
-            )
+            for transform, kwargs in augment:
+                dataset = dataset.map(
+                    lambda x, y: tf.cond(
+                        tf.random.uniform((1,)) > 0.5,
+                        true_fn=lambda: transform(x, y, **kwargs),
+                        false_fn=lambda: (x, y),
+                    ),
+                )
```

Comment:
@Aakanksha-Rana - this section does not behave in the same way if someone using `augment=False` in a script. it should retain the previous behavior and the warning should only appear if `augment==False`.

Reply:
`False is not None` will return `True` in python and thus the previous script which would have skipped this section would go into it. also someone using `augment=True` would fail. so this section should account and set appropriate defaults for those two use cases.

Reply:
Right. Thanks Satra. I will work it out.